pipeline/backend/kubernetes/pod.go

```diff
@@ -505,6 +514,18 @@ func startPod(ctx context.Context, engine *kube, step *types.Step, options Backe
 		return nil, err
 	}
 
+	if needsImagePullSecret(step) {
+		pullSecret, err := mkImagePullSecret(step, engineConfig, podName, engine.goos, options)
+		if err != nil {
+			return nil, err
+		}
+		log.Trace().Msgf("creating secret: %s", pullSecret.Name)
+		_, err = engine.client.CoreV1().Secrets(engineConfig.Namespace).Create(ctx, pullSecret, meta_v1.CreateOptions{})
```

Comment:
Make a function in `image_pull_secrets`.

Reply:
Done